pkg/cloudprovider/providers/vsphere/vsphere_util.go

```diff
@@ -326,8 +328,10 @@ func getcanonicalVolumePath(ctx context.Context, dc *vclib.Datacenter, volumePat
 			// It would fail and return an folder ID in the error message.
 			_, err := dc.GetVirtualDiskPage83Data(ctx, dummyDiskVolPath)
 			if err != nil {
-				re := regexp.MustCompile("File (.*?) was not found")
```

Comment:
Rather than match the error message at all, we can use the fault type like so:
```go
// fileNotFound returns the value of File if err is a FileNotFound soap fault, an empty string otherwise.
func fileNotFound(err error) string {
	if soap.IsSoapFault(err) {
		fault := soap.ToSoapFault(err)
		if f, ok := fault.VimFault().(types.FileNotFound); ok {
			return f.File
		}
	}
	return ""
}
```

Reply:
Note that this can be tested against vcsim, with the error returned here for example: https://github.com/kubernetes/kubernetes/blob/b66e332d3c19ce78e00dd7c904fa29e8f6784ba0/pkg/cloudprovider/providers/vsphere/vclib/datacenter_test.go#L138-L141